### PR TITLE
Issue ID 0004230 : Fixing a crash in chamfer command. Putting a null …

### PR DESCRIPTION
**Pull request by rajendrasp**
…check for the function return value

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, please confirm the following:

- [ ] Branch rebased on latest master `git pull --rebase upstream master`
- [ ] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [ ] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

### CODE REVIEW

File: src/Gui/Document.cpp [hunk 1]
```diff
@@ -786,7 +786,8 @@ void Document::slotChangedObject(const App::DocumentObject& Obj, const App::Prop
                     && d->_editingObject
                     && d->_editViewProviderParent 
                     && (Prop.isDerivedFrom(App::PropertyPlacement::getClassTypeId())
-                        || strstr(Prop.getName(),"Scale"))
+						// Issue ID 0004230 : getName() can return null in which case strstr() crashes
+						|| (Prop.getName() && strstr(Prop.getName(), "Scale")))
```
Comment (vocx-fc) [inline src/Gui/Document.cpp]: You added Tabs to indent this code. Please don't add Tabs to code that is already indented with spaces.

File: src/Mod/Sketcher/Gui/Command.cpp [hunk 2]
```diff
@@ -723,8 +723,9 @@ void CmdSketcherMirrorSketch::activated(int iMsg)
         std::vector<Part::Geometry *> tempgeo = tempsketch->getInternalGeometry();
         std::vector<Sketcher::Constraint *> tempconstr = tempsketch->Constraints.getValues();
 
-        std::vector<Part::Geometry *> mirrorgeo (tempgeo.begin()+addedGeometries+1,tempgeo.end());
-        std::vector<Sketcher::Constraint *> mirrorconstr (tempconstr.begin()+addedConstraints+1,tempconstr.end());
+		// If value of addedGeometries or addedConstraints is -1, it gets added to vector begin iterator and that is invlid
+		std::vector<Part::Geometry *> mirrorgeo(tempgeo.begin() + (addedGeometries + 1), tempgeo.end());
+		std::vector<Sketcher::Constraint *> mirrorconstr(tempconstr.begin() + (addedConstraints + 1), tempconstr.end());
```
Comment (vocx-fc) [inline src/Mod/Sketcher/Gui/Command.cpp]: Again, Tabs are added. Add spaces instead.

**Comment by rajendrasp** [comment]
It was default visual studio setting that I was using. I have made changes for spaces.

File: src/Mod/Sketcher/Gui/Command.cpp [hunk 3]
```diff
@@ -723,8 +723,9 @@ void CmdSketcherMirrorSketch::activated(int iMsg)
         std::vector<Part::Geometry *> tempgeo = tempsketch->getInternalGeometry();
         std::vector<Sketcher::Constraint *> tempconstr = tempsketch->Constraints.getValues();
 
-        std::vector<Part::Geometry *> mirrorgeo (tempgeo.begin()+addedGeometries+1,tempgeo.end());
-        std::vector<Sketcher::Constraint *> mirrorconstr (tempconstr.begin()+addedConstraints+1,tempconstr.end());
+		// If value of addedGeometries or addedConstraints is -1, it gets added to vector begin iterator and that is invlid
```
Comment (vocx-fc) [inline src/Mod/Sketcher/Gui/Command.cpp]: This comment still has tabs.

**Comment by rajendrasp** [comment]
My bad, it should be ok now

**Comment by luzpaz** [comment]
@rajendrasp thanks for submitting the PR. Next time do you mind discussing this on the forum so users can confirm you fix (and then linking to said forum thread)? Thanks. Also it helps to notate the [ticket](https://www.freecadweb.org/tracker/view.php?id=4230) with you intention to fix it. That way we don't have multiple devs doubling up on 1 ticket.

**Comment by rajendrasp** [comment]
Thanks @luzpaz for suggestions, will surely keep it in mind

**Comment by wwmayer** [comment]
Merged.

**Pull request closed without merging**